test/stream/producer_daemon_test.go

```diff
@@ -115,7 +115,9 @@ func (s *ProducerDaemonTestSuite) TestWriteData() {
 		t: s.T(),
 	}
 
-	app := application.Default(application.WithLoggerHandlers(handler))
+	app := application.Default(application.WithLoggerHandlers(handler), application.WithKernelExitHandler(func(code int) {
+		assert.Equal(s.T(), kernel.ExitCodeOk, code, "exit code should be %d", kernel.ExitCodeOk)
```

Comment:
Maybe a short-hand exit code assertion helps to not duplicate this line over and over again :D

Reply:
Could be, but as this code is already being refactored, I will leave it like this for now :)